src/elements/e-video.vue

```diff
@@ -0,0 +1,254 @@
+ *    - vimeo not able to disable controls
+ *  size:
+ *    - width & height can either be responsive or hard coded
+ *    - youtube standard -> width="560", height="315"
+ */
+  import { PropType, defineComponent } from 'vue';
+
+  const videoUrlSources = {
+    youtube: 'https://www.youtube.com',
+    vimeo: 'https://vimeo.com/',
+  };
+
+  const playerUrlSources = {
+    youtube: 'https://www.youtube.com/embed/',
+    vimeo: 'https://player.vimeo.com/video/',
+  };
+
+  export default defineComponent({
+    name: 'e-video',
+    status: 0, // TODO: remove when component was prepared for current project.
+
+    // components: {},
+
+    props: {
+      /**
+       * Allows the user to watch the video in fullscreen
+       */
+      allowfullscreen: {
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Specifies the source of the video
+       *
+       * We can decide between youtube and vimeo
+       */
+      source: {
+        type: String as PropType<'youtube' | 'vimeo'>,
+        required: true,
+        validator(value: string) {
+          return Object.keys(playerUrlSources).includes(value);
+        },
+      },
+
+      /**
+       * Allows to add/remoe the border around the player
+       *
+       * When set to 0 the border is completely removed
+       *
+       * Default from iframe is 2
+       */
+      frameborder: {
+        type: [String, Number],
+        default: 2,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       */
+      width: {
+        type: [String, Number],
+        default: 560,
+      },
+
+      /**
+       * Allows to set the video player height.
+       */
+      height: {
+        type: [String, Number],
+        default: 315,
+      },
+
+      /**
+       * Defines the players size to be responsive 100% width to the parent container
+       */
+      responsive: {
+        type: Boolean,
+        default: false,
+      },
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = playerUrlSources[this.source];
+
+        if (!(this.videoId || this.videoUrl)) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
+        }
+
+        if (this.videoUrl && !this.videoId) {
+          if (!this.validateUrl()) {
+            return undefined;
+          }
+
+          if (this.source === 'youtube') {
+            /**
+             * Extract id from video url
+             */
+            const id = this.videoUrl.substring(
```

Comment:
todo: please use URLSearchParams instead. See https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/get

Reply:
Or, maybe easier to do: use `String.startsWith()` instead and check if the url starts with one of the provided ones. This would allow to add additional url params, but I don't think this would be a critical case. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith?retiredLocale=de